robot/basestation/static/js/elements/navigation/localization.js

```diff
@@ -0,0 +1,182 @@
+/*
+This script uses a lot of p5.js functions and concepts.
+I recommend using the reference: https://p5js.org/reference/#/p5/
+before doing any work on this script as it will greatly facilitate the process.
+*/
+
+$(document).ready(() => {
+    myp5 = new p5(sketch)
+})
+
+let sketch = function(sketch) {
+    roverHeading = 0
+    roverX = 0
+    roverY = 0
+    dragDiffX = 0
+    dragDiffY = 0
+    zoomValue = 1
+    keys = { //javascript keycodes
+        'r': 82,
+        'w': 87,
+        'a': 65,
+        's': 83,
+        'd': 68
+    }
+    toggle = false
+
+    MAX_ZOOM = 2
+    MIN_ZOOM = 0.15
+    CANVAS_HEIGHT = 300
+    CANVAS_WIDTH = 450
+    baseLat = 0
+    baseLong = 0
+    rover_position_listener.subscribe(function(message) {
+        roverLong = message.x
+        roverLat = message.y
+        roverHeading = message.z
+        adjustValuesToCartesian(roverLat, roverLong)
+    })
+
+    sketch.setup = function() {
+        canvas = sketch.createCanvas(CANVAS_WIDTH, CANVAS_HEIGHT)
+        canvas.parent('#canvasContainer')
+        sketch.angleMode(sketch.DEGREES) // makes p5.js use DEG instead of RAD
+    }
+
+    sketch.draw = function() {
+        if (sketch.keyIsPressed == true) {
+            controls()
+        }
+
+        sketch.background(225)
+        drawResizeButton()
+        sketch.translate(sketch.width / 2 + dragDiffX, sketch.height / 2 + dragDiffY)
+        sketch.scale(zoomValue)
+        sketch.stroke(200)
+        sketch.fill(120)
+        drawGrid()
+        drawBase()
+        drawRover()
+    }
+
+    /* ----------------------------------------
+    FUNCTIONS THAT DRAW THINGS
+    */
+
+    function drawBase() {
+        sketch.push()
+        sketch.scale(1 / zoomValue)
+        // (x, y, diameter)
+        sketch.circle(0, 0, 10)
+        sketch.pop()
+    }
+
+    function drawRover() {
+        sketch.push()
+        sketch.translate(roverX, roverY)
+        sketch.rotate(roverHeading)
+        sketch.scale(1 / zoomValue)
+        // 3 points (x, y, x, y, x ,y)
+        sketch.triangle(0, -15, 10, 10, -10, 10)
+        sketch.pop()
+    }
+
+    function drawGrid() {
+        for (let x = -10000; x < 10000; x += 100) {
+            sketch.line(x, -10000, x, 10000)
+            sketch.text(x, x + 1, 12)
+        }
+
+        for (let y = -10000; y < 10000; y += 100) {
+            sketch.line(-10000, y, 10000, y)
+            sketch.text(y, 1, y + 12)
```

Comment:
Im assuming the canvas is actually 20 000 x 20 000 of some units? Is that right? Will this maybe change? If so consider putting these values in variables/constants